### PR TITLE
Added shortcut methods for controllers

### PR DESCRIPTION
**Pull request by Cydonia7**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes (symfony/symfony#11593) |
| Applies to | 2.6+ |
| Fixed tickets | #4666 |

This commit is associated to https://github.com/symfony/symfony/pull/11593 that adds new shortcut methods to controllers.

If anything is wrong with my commit or this description, please tell me and I will fix it as soon as possible. I'm glad I finally try to help this awesome project.

### CODE REVIEW

**Comment by Cydonia7** [comment]
@weaverryan I think I've changed it all, would you please take a look?

File: book/security.rst [hunk 1]
```diff
@@ -239,6 +239,75 @@ user to be logged in to access this URL:
...
+~~~~~~~~~~~~~~~~~~~~~
+
+Protecting your application based on URL patterns is easy, but may not be
+fine-grained enough in certain cases. When necessary, you can easily force
+authorization from inside a controller::
+
+.. versionadded:: 2.6
+    The ``denyAccessUnlessGranted()`` method was introduced in Symfony 2.6. Previously (and
+    still now), you could check access directly and throw the ``AccessDeniedException`` as shown
+    in the example below).
```
Comment (xabbuh) [inline book/security.rst]: I would move it above the paragraph (right below the "Securing a Controller" headline).

File: book/security.rst [hunk 2]
```diff
@@ -239,6 +239,75 @@ user to be logged in to access this URL:
...
+
+    // ...
+
+    public function helloAction($name)
+    {
+        $this->denyAccessUnlessGranted('ROLE_ADMIN', null, 'Unable to access this page!');
```
Comment (javiereguiluz) [inline book/security.rst]: Maybe we should add a brief explanation of the weird `null` middle argument?

File: book/security.rst [hunk 3]
```diff
@@ -253,6 +322,7 @@ user to be logged in to access this URL:
                 array('path' => '^/admin', 'role' => 'ROLE_ADMIN'),
             ),
         ));
+>>>>>>> master
```
Comment (javiereguiluz) [inline book/security.rst]: This line should be removed.

File: quick_tour/the_controller.rst [hunk 4]
```diff
@@ -192,6 +192,7 @@ method::
 
 The ``redirectToRoute()`` method takes as arguments the route name and an optional
 array of parameters and redirects the user to the URL generated with those arguments.
+>>>>>>> master
```
Comment (javiereguiluz) [inline quick_tour/the_controller.rst]: This line should be removed.

File: book/security.rst (hunk 2 above)
Comment (Cydonia7) [inline book/security.rst]: That's right.

File: book/security.rst (hunk 3 above)
Comment (Cydonia7) [inline book/security.rst]: I can't believe I forgot it.

**Comment by Cydonia7** [comment]
I fixed the issues @xabbuh @javiereguiluz

File: book/security.rst [hunk 5]
```diff
@@ -239,6 +239,77 @@ user to be logged in to access this URL:
...
+:doc:`FrameworkExtraBundle documentation </bundles/SensioFrameworkExtraBundle/annotations/security>`.
+
+Securing other Services
+~~~~~~~~~~~~~~~~~~~~~~~
+
+In fact, anything in Symfony can be protected using a strategy similar to
+the one seen in the previous section. For example, suppose you have a service
+(i.e. a PHP class) whose job is to send emails from one user to another.
+You can restrict use of this class - no matter where it's being used from -
+to users that have a specific role.
```
Comment (xabbuh) [inline book/security.rst]: The following code block doesn't really belong to this paragraph.

**Comment by javiereguiluz** [comment]
Could we made a final effort to complete this PR? It would be nice to show and use the new shortcuts, but I understand that the PR is so big and complex that is taking forever to complete :(  If there is much work to do, could we create another separate PR?

**Comment by Cydonia7** [comment]
What is needed to "complete" it ?

**Comment by weaverryan** [comment]
Ah, we're clearly waiting on me only! I'll put this on my primary list and so will review and merge very soon. Sorry for being in the way here!

File: book/controller.rst [hunk 6]
```diff
@@ -430,24 +430,34 @@ Redirecting
...
+        // redirectToRoute is equivalent to using redirect() and generateUrl() together:
+        // return $this->redirect($this->generateUrl('homepage'), 301);
     }
 
-The ``generateUrl()`` method is just a helper function that generates the URL
-for a given route. For more information, see the :doc:`Routing </book/routing>`
-chapter.
+.. versionadded:: 2.6
+    The ``redirectToRoute()`` method was added in Symfony 2.6. Previously (and still now), you
+    could use ``redirect()`` and ``generateUrl()`` together for this (see the example below).
```
Comment (weaverryan) [inline book/controller.rst]: `(see the example above)`

**Comment by weaverryan** [comment]
Hey @Cydonia7!

Sorry for the delay on my end. I just need *one* more thing from you. I added a really easy comment, but what I really need is a rebase. I need to merge this into the 2.6 branch, which is conflicting in the security chapter. You can kind of see it here - the changes to `security.rst` are too big. If you search for `_book-security-securing-controller`, you'll see that after your commits, there are 2 of these sections. This is due to a big refatoring of the security chapter, so it's my fault for making your life difficult.

Would you be able to rebase? The result should be that `security.rst` has less changes (i.e. it doesn't look like you added a new section).

Thanks!

**Comment by Cydonia7** [comment]
I tried and didn't manage to do it for now... The rebase is so complex I can't even understand what I am doing sometimes (I've had to do like 5 times rebase --continue and changing commits I don't even authored).

So what should I do ?

**Comment by xabbuh** [comment]
@Cydonia7 If you were able to join #symfony-docs on IRC, someone could probably be able to help you. Otherwise, if you don't mind, we can take your changes, rebase them and create a new PR for you.

**Comment by Cydonia7** [comment]
I won't have time during the next week to do this but I don't mind, you can create a new PR with my changes.

**Comment by WouterJ** [comment]
Okay, I'll try :)

**Comment by Cydonia7** [comment]
Thanks @WouterJ, I didn't think I would have such a hard time rebasing this...

**Comment by stof** [comment]
@Cydonia7 if you are changing commits you have not authored, it means you are doing the wrong rebase. You should rebase your own feature branch, not the symfony branch.

**Comment by WouterJ** [comment]
Closing in favor of #5046 (and indeed, it was a hell of a rebase, literally every change in this PR was conflicting...)

**Pull request closed without merging**